vue/require-default-prop not picking up property name

<!--
  Before posting the issue, please confirm that the problem you're getting
  is not related with your code editor configuration.
  To make sure it's not, run: yarn eslint src/your-file.vue
-->

**Tell us about your environment**	
* **ESLint version:**	5.12.0
* **eslint-plugin-vue version:** 5.1.0
* **Node version:**	8.11.4

**Please show your full configuration:**
<!-- Paste content of your .eslintrc file -->
```javascript
module.exports = {
  root: true,
  env: {
    browser: true,
    node: true
  },
  parserOptions: {
    parser: 'babel-eslint'
  },
  extends: [
    'standard',
    'plugin:vue/recommended'
  ],
  // required to lint *.vue files
  plugins: [
    'vue'
  ],
  // add your custom rules here
  rules: {
    'no-console':  process.env.NODE_ENV === 'production' ? 'error' : 'off',
    'no-debugger': process.env.NODE_ENV === 'production' ? 'error' : 'off',

    "no-multi-spaces": "off",
    "key-spacing": "off",
    "comma-dangle": ["warn", "always-multiline"],

    "vue/max-attributes-per-line": ["warn", {
      "singleline": 3,
      "multiline": {
        "max": 1,
        "allowFirstLine": false
      }
    }],
  },
  globals: {
    CORE_API_BASE_URL: false,
  },
}
```

**What did you do?**	
<!-- Please include the actual source code causing the issue. -->
```vue
<script>
export default {
	props: {
	    tallStyle: Object,
	    'inverse': Boolean,
	    'actions': Array,
	}
}
</script>
```

**What did you expect to happen?**

The warnings should show the property names "tallStyle" and "actions"

**What actually happened?**
<!-- Please include the actual, raw output from ESLint. -->

```
  95:5   warning  Prop 'tallStyle' requires default value to be set  vue/require-default-prop
  97:5   warning  Prop 'undefined' requires default value to be set  vue/require-default-prop
```

--- a/lib/rules/require-default-prop.js
+++ b/lib/rules/require-default-prop.js
@@ -141,11 +141,13 @@ module.exports = {
       const propsToReport = excludeBooleanProps(propsWithoutDefault)
 
       for (const prop of propsToReport) {
+        const propName = prop.propName != null ? prop.propName : `[${context.getSourceCode().getText(prop.key)}]`
+
         context.report({
           node: prop.node,
           message: `Prop '{{propName}}' requires default value to be set.`,
           data: {
-            propName: prop.key.name
+            propName
           }
         })
       }

--- a/lib/rules/require-valid-default-prop.js
+++ b/lib/rules/require-valid-default-prop.js
@@ -109,11 +109,12 @@ module.exports = {
         const defType = getValueType(def.value)
         if (!defType || typeNames.has(defType)) continue
 
+        const propName = prop.propName != null ? prop.propName : `[${context.getSourceCode().getText(prop.key)}]`
         context.report({
           node: def,
           message: "Type of the default value for '{{name}}' prop must be a {{types}}.",
           data: {
-            name: prop.key.name,
+            name: propName,
             types: Array.from(typeNames).join(' or ').toLowerCase()
           }
         })

--- a/lib/utils/index.js
+++ b/lib/utils/index.js
@@ -484,13 +484,16 @@ module.exports = {
       props = propsNode.value.properties
         .filter(prop => prop.type === 'Property')
         .map(prop => {
-          return { key: prop.key, value: this.unwrapTypes(prop.value), node: prop }
+          return {
+            key: prop.key, value: this.unwrapTypes(prop.value), node: prop,
+            propName: this.getStaticPropertyName(prop)
+          }
         })
     } else {
       props = propsNode.value.elements
         .map(prop => {
           const key = prop.type === 'Literal' && typeof prop.value === 'string' ? prop : null
-          return { key, value: null, node: prop }
+          return { key, value: null, node: prop, propName: key != null ? prop.value : null }
         })
     }
 

--- a/tests/lib/rules/require-default-prop.js
+++ b/tests/lib/rules/require-default-prop.js
@@ -231,6 +231,57 @@ ruleTester.run('require-default-prop', rule, {
         message: `Prop 'a' requires default value to be set.`,
         line: 4
       }]
+    },
+
+    // computed propertys
+    {
+      filename: 'test.vue',
+      code: `
+        export default {
+          props: {
+            a: String,
+            'b': String,
+            ['c']: String,
+            [\`d\`]: String,
+          }
+        };
+      `,
+      errors: [{
+        message: `Prop 'a' requires default value to be set.`,
+        line: 4
+      }, {
+        message: `Prop 'b' requires default value to be set.`,
+        line: 5
+      }, {
+        message: `Prop 'c' requires default value to be set.`,
+        line: 6
+      }, {
+        message: `Prop 'd' requires default value to be set.`,
+        line: 7
+      }]
+    },
+    // unknown static name
+    {
+      filename: 'test.vue',
+      code: `
+        export default {
+          props: {
+            [foo]: String,
+            [bar()]: String,
+            [baz.baz]: String,
+          }
+        };
+      `,
+      errors: [{
+        message: `Prop '[foo]' requires default value to be set.`,
+        line: 4
+      }, {
+        message: `Prop '[bar()]' requires default value to be set.`,
+        line: 5
+      }, {
+        message: `Prop '[baz.baz]' requires default value to be set.`,
+        line: 6
+      }]
     }
   ]
 })

--- a/tests/lib/rules/require-valid-default-prop.js
+++ b/tests/lib/rules/require-valid-default-prop.js
@@ -444,6 +444,37 @@ ruleTester.run('require-valid-default-prop', rule, {
       parserOptions: { ecmaVersion: 6, sourceType: 'module' },
       parser: 'typescript-eslint-parser',
       errors: errorMessage('function or number')
+    },
+
+    {
+      filename: 'test.vue',
+      code: `export default {
+        props: {
+          'foo': {
+            type: Object,
+            default: ''
+          },
+          ['bar']: {
+            type: Object,
+            default: ''
+          },
+          [baz]: {
+            type: Object,
+            default: ''
+          }
+        }
+      }`,
+      parserOptions,
+      errors: [{
+        message: `Type of the default value for 'foo' prop must be a function.`,
+        line: 5
+      }, {
+        message: `Type of the default value for 'bar' prop must be a function.`,
+        line: 9
+      }, {
+        message: `Type of the default value for '[baz]' prop must be a function.`,
+        line: 13
+      }]
     }
   ]
 })

